app/scripts/services/validateRegistrant.js

```diff
@@ -26,9 +25,9 @@ angular.module('confRegistrationWebApp')
           }
         }
         if ((ruleOperand === 'OR' && validRuleCount > 0) || (ruleOperand === 'AND' && validRuleCount <= i)) {
-          break;
+          return false; // Exit lodash foreach as we found a case which determines the whole outcome of this function
```

Comment:
`return false;` (`return;` would probably work too) exits the for loop (which is a function) early and not the entire function. Lodash wraps the iterations in a function.

I did have a version that I suggested to the contractors where you returned values right there but they didn't like it. I was trying to change as few things as possible to reduce merge conflicts.

You did just make me notice that it is relying on `i` in `validRuleCount <= i`. So don't merge it. Let me fix that. Good catch.